sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -220,7 +222,16 @@ def _batch_send(
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()
+            parts = response.parts()
```

Comment:
You need to consume the iterator (I know today it's a list, but you're not supposed to know)
```python
parts = list(response.parts())
```